xapian-letor/api/featurelist.cc

```diff
@@ -40,12 +41,14 @@ FeatureList::FeatureList()
     feature.push_back(new CollTfCollLenFeature());
     feature.push_back(new TfIdfDoclenFeature());
     feature.push_back(new TfDoclenCollTfCollLenFeature());
+    internal = new FeatureList::Internal();
```

Comment:
It's cleaner to assign the internals using a member initialiser list (i.e. `    : internal(new FeatureList::Internal())` before the opening `{` of the constructor.

If you consistently initialise the internal pointer like that in all the constructors, then you can rely on it being set in other methods.  Similarly below.